*NB : Voir d'abord stat_descriptives_youtube.ipynb* 

**On cherche si des livres et auteurs de nos bases de données de livres sont mentionnés dans des vidéos de nos bases de données youtube. Pour cela, on étudie les sous-titres de ces vidéos youtube.**

# 1 Recherche des titres de livres dans les descriptions de vidéos youtube

In [5]:
#On importe la bibliothèque pandas
import pandas as pd
import matplotlib.pyplot as plt

def livre_dans_sous_titres(livre,youtube):
    #On charge les tables dont on a besoin",
    books_data = pd.read_csv(f"{livre}.csv", encoding="utf-8")
    youtube_data = pd.read_csv(f"{youtube}.csv", encoding="utf-8")
    print("Livres de la base de données "f'{livre}.csv'" dans les sous-titres de vidéos youtube de la base de données "f'{youtube}.csv')
    # On convertie la colonne 'Subtitles'et 'Titre' en type string
    youtube_data['Subtitles'] = youtube_data['Subtitles'].astype(str)
    books_data['Titre'] = books_data['Titre'].astype(str)
    # On initialise une nouvelle liste pour stocker les résultats
    result = []
    # On itére sur chaque liste de la colonne 'Titre' de books_data
    for index1, row1 in books_data.iterrows():
        count_list = []  # Liste pour stocker les résultats de cette ligne
        # On itére sur chaque liste de la colonne 'Subtitles' de youtube_data
        for index2, row2 in youtube_data.iterrows():
            # On vérifie si la liste de books_data (row1['Titre']) est une sous-liste de la liste de youtube_data (row2['Subtitles'])
            if row1['Titre'] in row2['Subtitles']:
                count_list.append(1)  # Si elle est incluse, compter une occurrence
            else:
                count_list.append(0)  # Sinon, aucune occurrence
        # On ajoute la liste des Titres, des Auteurs de books_data et la somme des occurrences
        result.append([row1['Titre'], row1['Auteur'], sum(count_list)])
    # On crée la nouvelle DataFrame contenant les résultats
    colonnes=['Titres','Auteurs','Occurrences dans les sous-titres de vidéos']
    result_df = pd.DataFrame(result, columns = colonnes )
    # On filtre les lignes où 'Occurrences dans les sous-titres des vidéos' est différente de 0 (On ne s'intéresse qu'aux livres cités au moins une fois)
    filtered_df = result_df[result_df['Occurrences dans les sous-titres de vidéos'] != 0]
    # On affiche le DataFrame filtré
    return filtered_df

def livre_dans_sous_titres2(livre,youtube):
    #On charge les tables dont on a besoin",
    books_data = pd.read_csv(f"{livre}.csv", encoding="utf-8")
    youtube_data = pd.read_csv(f"{youtube}.csv", encoding="utf-8")
    print("Livres de la base de données "f'{livre}.csv'" dans les sous-titres de vidéos youtube de la base de données "f'{youtube}.csv')
    # On convertie la colonne 'Subtitles'et 'Titre' en type string
    youtube_data['Subtitles'] = youtube_data['Subtitles'].astype(str)
    books_data['titre'] = books_data['titre'].astype(str)
    # On initialise une nouvelle liste pour stocker les résultats
    result = []
    # On itére sur chaque liste de la colonne 'titre' de books_data
    for index1, row1 in books_data.iterrows():
        count_list = []  # Liste pour stocker les résultats de cette ligne
        # On itére sur chaque liste de la colonne 'Subtitles' de youtube_data
        for index2, row2 in youtube_data.iterrows():
            # On vérifie si la liste de books_data (row1['titre']) est une sous-liste de la liste de youtube_data (row2['Subtitles'])
            if row1['titre'] in row2['Subtitles']:
                count_list.append(1)  # Si elle est incluse, compter une occurrence
            else:
                count_list.append(0)  # Sinon, aucune occurrence
        # On ajoute la liste des titres, des auteurs de books_data et la somme des occurrences
        result.append([row1['titre'], row1['auteur'], sum(count_list)])
    # On crée la nouvelle DataFrame contenant les résultats
    colonnes=['Titres','Auteurs','Occurrences dans les sous-titres de vidéos']
    result_df = pd.DataFrame(result, columns = colonnes )
    # On filtre les lignes où 'Occurrences dans les sous-titres des vidéos' est différente de 0 (On ne s'intéresse qu'aux livres cités au moins une fois)
    filtered_df = result_df[result_df['Occurrences dans les sous-titres de vidéos'] != 0]
    # On affiche le DataFrame filtré
    return filtered_df

## 1.1 Recherche des titres de livre dans la base de données ss_youtube_livre_2019_2023.csv

### 1.1.1 Recherche des titres de livre de la base de données 23_livres_les_plus_lus_babelio.csv

In [ ]:
livre_dans_sous_titres('23_livres_les_plus_lus_babelio','ss_youtube_livre_2019_2023')

Il n'y a aucun livre de cette base de données mentionné.

### 1.1.2 Recherche des titres de livre de la base de données best_sellers_fnac_2023_cleaned.csv

In [ ]:
livre_dans_sous_titres('best_sellers_fnac_2023_cleaned','ss_youtube_livre_2019_2023')

Il n'y a aucun livre de cette base de données mentionné.

### 1.1.3 Recherche des titres de livre de la base de données livres_les_plus_empruntes_a_paris.csv

In [ ]:
livre_dans_sous_titres2('livres_les_plus_empruntes_a_paris','ss_youtube_livre_2019_2023')

**Le livre *Noa* de Marc Lévy apparait dans les sous-titres de deux vidéos.** Il y a donc un livre parmi les livres les plus empruntés à Paris qui est dans les sous-titres de deux vidéos. 

### 1.1.4 Recherche des titres de livre de la base de données livraddict_prix_2024.csv

In [ ]:
livre_dans_sous_titres('livraddict_prix_2024','ss_youtube_livre_2019_2023')

**Les livres *Babel* de R. F. Kuang et *Papa* de Hélène Delforge et Quentin Gréban apparaissent chacun à deux reprises dans les sous-titres de vidéos youtube.** Ces livres qui sont deux des livres les mieux notés par les utilisateurs de livraddict parmi les livres sortis en 2023, sont chacun dans deux sous-titres de vidéos.

### 1.1.5 Recherche des titres de livre de la base de données prix_litteraires.csv

In [14]:
livre_dans_sous_titres('prix_litteraires','ss_youtube_livre_2019_2023')

Livres de la base de données prix_litteraires.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Titres           Auteurs  Occurrences dans les sous-titres de vidéos
105  Révolution  Florent Grouazel                                           3
108       Fritz     Coline Pierré                                           1
191         555    Hélène Gestern                                          86

**Les livres *Révolution* de Florent Grouazel, *Fritz* de Coline Pierré et *555* de Hélène Gestern apparaissent dans les sous-titres de vidéos youtube.** Ces 3 livres qui ont reçu un prix littéraire entre 2019 et 2023 sont donc dans des sous-titres de vidéos.

### 1.1.6 Recherche des titres de livre de la base de données Base_finale.csv

In [ ]:
livre_dans_sous_titres('Base_finale','ss_youtube_livre_2019_2023')

Livres de la base de données Base_finale.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Titres</th>
      <th>Auteurs</th>
      <th>Occurrences dans les sous-titres de vidéos</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>555</td>
      <td>Hélène Gestern</td>
      <td>86</td>
    </tr>
    <tr>
      <th>8</th>
      <td>argonne</td>
      <td>Stéphane Emond</td>
      <td>1</td>
    </tr>
    <tr>
      <th>19</th>
      <td>avant que j'oublie</td>
      <td>Anne Pauly</td>
      <td>1</td>
    </tr>
    <tr>
      <th>21</th>
      <td>babel</td>
      <td>R. F. Kuang</td>
      <td>1</td>
    </tr>
    <tr>
      <th>31</th>
      <td>blizzard</td>
      <td>Marie Vingtras</td>
      <td>2</td>
    </tr>
    <tr>
      <th>44</th>
      <td>cent mille ans</td>
      <td>Gaspard d’Allens</td>
      <td>1</td>
    </tr>
    <tr>
      <th>49</th>
      <td>chavirer</td>
      <td>Lola Lafon</td>
      <td>1</td>
    </tr>
    <tr>
      <th>61</th>
      <td>conte de fées</td>
      <td>Stephen King</td>
      <td>2</td>
    </tr>
    <tr>
      <th>80</th>
      <td>des vivants</td>
      <td>Meltz,  Raphaël</td>
      <td>4</td>
    </tr>
    <tr>
      <th>101</th>
      <td>fritz</td>
      <td>Coline Pierré</td>
      <td>1</td>
    </tr>
    <tr>
      <th>117</th>
      <td>héritage</td>
      <td>Miguel Bonnefoy</td>
      <td>18</td>
    </tr>
    <tr>
      <th>122</th>
      <td>j'ai oublié</td>
      <td>Bulle Ogier</td>
      <td>8</td>
    </tr>
    <tr>
      <th>139</th>
      <td>l'escalade</td>
      <td>Dugomier</td>
      <td>1</td>
    </tr>
    <tr>
      <th>146</th>
      <td>la chaleur</td>
      <td>Victor Jestin</td>
      <td>21</td>
    </tr>
    <tr>
      <th>150</th>
      <td>la colère</td>
      <td>S. A. Cosby</td>
      <td>34</td>
    </tr>
    <tr>
      <th>159</th>
      <td>la décision</td>
      <td>Tuil,  Karine</td>
      <td>17</td>
    </tr>
    <tr>
      <th>160</th>
      <td>la dépendance</td>
      <td>Rachel Cusk</td>
      <td>5</td>
    </tr>
    <tr>
      <th>161</th>
      <td>la faille</td>
      <td>Franck Thilliez</td>
      <td>1</td>
    </tr>
    <tr>
      <th>171</th>
      <td>la grâce</td>
      <td>Thibault de Montaigu</td>
      <td>13</td>
    </tr>
    <tr>
      <th>173</th>
      <td>la maison</td>
      <td>Emma Becker</td>
      <td>62</td>
    </tr>
    <tr>
      <th>190</th>
      <td>la sentence</td>
      <td>Louise Erdrich</td>
      <td>5</td>
    </tr>
    <tr>
      <th>193</th>
      <td>la tentation</td>
      <td>Luc Lang</td>
      <td>8</td>
    </tr>
    <tr>
      <th>198</th>
      <td>la vraie vie</td>
      <td>Adeline Dieudonné</td>
      <td>8</td>
    </tr>
    <tr>
      <th>202</th>
      <td>le bon sens</td>
      <td>Michel Bernard</td>
      <td>4</td>
    </tr>
    <tr>
      <th>207</th>
      <td>le consentement</td>
      <td>Vanessa Springora</td>
      <td>2</td>
    </tr>
    <tr>
      <th>209</th>
      <td>le culte</td>
      <td>Camilla Läckberg, Henrik Fexeus</td>
      <td>2</td>
    </tr>
    <tr>
      <th>227</th>
      <td>le lâche</td>
      <td>Jarred McGinnis</td>
      <td>4</td>
    </tr>
    <tr>
      <th>236</th>
      <td>le retournement</td>
      <td>Manuel Carcassonne</td>
      <td>1</td>
    </tr>
    <tr>
      <th>262</th>
      <td>les cinq</td>
      <td>Mathieu Rochelle</td>
      <td>16</td>
    </tr>
    <tr>
      <th>264</th>
      <td>les conditions idéales</td>
      <td>Mokhtar Amoudi</td>
      <td>2</td>
    </tr>
    <tr>
      <th>288</th>
      <td>les passagers</td>
      <td>Gilles Verdet</td>
      <td>1</td>
    </tr>
    <tr>
      <th>293</th>
      <td>les possibles</td>
      <td>Grimaldi,  Virginie</td>
      <td>2</td>
    </tr>
    <tr>
      <th>303</th>
      <td>les éclats</td>
      <td>Bret Easton Ellis</td>
      <td>2</td>
    </tr>
    <tr>
      <th>330</th>
      <td>mon territoire</td>
      <td>Tess Sharpe</td>
      <td>4</td>
    </tr>
    <tr>
      <th>348</th>
      <td>nature humaine</td>
      <td>Serge Joncour</td>
      <td>1</td>
    </tr>
    <tr>
      <th>351</th>
      <td>noa</td>
      <td>Lévy,  Marc</td>
      <td>4</td>
    </tr>
    <tr>
      <th>370</th>
      <td>panorama</td>
      <td>Lilia Hassaine</td>
      <td>2</td>
    </tr>
    <tr>
      <th>371</th>
      <td>papa</td>
      <td>Hélène Delforge et Quentin Gréban</td>
      <td>34</td>
    </tr>
    <tr>
      <th>377</th>
      <td>performance</td>
      <td>Simon Liberati</td>
      <td>14</td>
    </tr>
    <tr>
      <th>399</th>
      <td>révolution</td>
      <td>Florent Grouazel</td>
      <td>21</td>
    </tr>
    <tr>
      <th>400</th>
      <td>s'adapter</td>
      <td>Clara Dupont-Monod</td>
      <td>9</td>
    </tr>
    <tr>
      <th>401</th>
      <td>sa préférée</td>
      <td>Sarah Jollien-Fardel</td>
      <td>1</td>
    </tr>
    <tr>
      <th>402</th>
      <td>sabre</td>
      <td>Emmanuel Ruben</td>
      <td>7</td>
    </tr>
    <tr>
      <th>423</th>
      <td>tenir sa langue</td>
      <td>Polina Panassenko</td>
      <td>1</td>
    </tr>
    <tr>
      <th>432</th>
      <td>tombeaux</td>
      <td>Annette Wieviorka</td>
      <td>2</td>
    </tr>
    <tr>
      <th>443</th>
      <td>tueurs</td>
      <td>Jean-Michel Espitallier</td>
      <td>1</td>
    </tr>
    <tr>
      <th>452</th>
      <td>un miracle</td>
      <td>Mas,  Victoria</td>
      <td>5</td>
    </tr>
    <tr>
      <th>458</th>
      <td>une belle vie</td>
      <td>Virginie Grimaldi</td>
      <td>2</td>
    </tr>
    <tr>
      <th>476</th>
      <td>western</td>
      <td>Maria Pourchet</td>
      <td>3</td>
    </tr>
    <tr>
      <th>482</th>
      <td>à la ligne</td>
      <td>Joseph Ponthus</td>
      <td>2</td>
    </tr>
    <tr>
      <th>485</th>
      <td>à tout jamais</td>
      <td>Colleen Hoover</td>
      <td>12</td>
    </tr>
  </tbody>
</table>
</div>

In [ ]:
df = livre_dans_sous_titres('Base_finale','ss_youtube_livre_2019_2023')
df.shape # connaitre le nombre de lignes de la base de données obtenue

**51 livre appraissent dans les sous-titres de vidéos youtube.** Le nombre d'occurrences varie entre 1 et 86.

## 1.2 Recherche des titres de livre dans la base de données ss_youtube_nouveauté_livre_2023.csv

### 1.2.1 Recherche des titres de livre de la base de données 23_livres_les_plus_lus_babelio.csv

In [ ]:
livre_dans_sous_titres('23_livres_les_plus_lus_babelio','ss_youtube_nouveauté_livre_2023')

Il n'y a aucun livre de cette base de données mentionné.

### 1.2.2 Recherche des titres de livre de la base de données best_sellers_fnac_2023_cleaned.csv

In [ ]:
livre_dans_sous_titres('best_sellers_fnac_2023_cleaned','ss_youtube_nouveauté_livre_2023')

**Le livre *Icebreaker* de Hannah Grace apparaît dans une description de vidéo youtube.** Un des best sellers de la Fnac en 2023 est dans sous-titres d'une vidéo youtube.

### 1.2.3 Recherche des titres de livre de la base de données livres_les_plus_empruntes_a_paris.csv

In [ ]:
livre_dans_sous_titres2('livres_les_plus_empruntes_a_paris','ss_youtube_nouveauté_livre_2023')

**Les livres *Noa* de Marc Lévy et *La maison* de Emma Becker apparaissent dans les sous-titres de plusieurs vidéos.** Il y a donc deux livres parmi les livres les plus empruntés à Paris qui sont dans les sous-titres de vidéos. 

### 1.2.4 Recherche des titres de livre de la base de données livraddict_prix_2024.csv

In [ ]:
livre_dans_sous_titres('livraddict_prix_2024','ss_youtube_nouveauté_livre_2023')

**Les livres *Wonka* de Sibéal Pounder, *La colère* de S. A. Cosby, *Babel* de R. F. Kuang, *Papa* de Hélène Delforge et Quentin Gréban apparaissent  dans les sous-titres de vidéos youtube.** Ces livres qui sont quatre des livres les mieux notés par les utilisateurs de livraddict parmi les livres sortis en 2023, sont chacun dans des sous-titres de vidéos.

### 1.2.5 Recherche des titres de livre de la base de données prix_litteraires.csv

In [ ]:
livre_dans_sous_titres('prix_litteraires','ss_youtube_nouveauté_livre_2023')

**Les livres *J'ai oublié* de Bulle Ogier, *Les Patriotes* de Sana Krasikov, *Betty* de Arnaldur Indridason, *Révolution* de Florent Grouazel, *Paul Morand* de Pauline Dreyfus et *555* de Hélène Gestern apparaissent dans les sous-titres de vidéos youtube.** Ces 6 livres qui ont reçu un prix littéraire entre 2019 et 2023 sont donc dans des sous-titres de vidéos.

### 1.2.6 Recherche des titres de livre de la base de données Base_finale.csv

In [ ]:
livre_dans_sous_titres('Base_finale','ss_youtube_nouveauté_livre_2023')

Livres de la base de données Base_finale.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv

<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Titres</th>
      <th>Auteurs</th>
      <th>Occurrences dans les sous-titres de vidéos</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>1</th>
      <td>555</td>
      <td>Hélène Gestern</td>
      <td>127</td>
    </tr>
    <tr>
      <th>31</th>
      <td>blizzard</td>
      <td>Marie Vingtras</td>
      <td>1</td>
    </tr>
    <tr>
      <th>80</th>
      <td>des vivants</td>
      <td>Meltz,  Raphaël</td>
      <td>1</td>
    </tr>
    <tr>
      <th>117</th>
      <td>héritage</td>
      <td>Miguel Bonnefoy</td>
      <td>17</td>
    </tr>
    <tr>
      <th>122</th>
      <td>j'ai oublié</td>
      <td>Bulle Ogier</td>
      <td>22</td>
    </tr>
    <tr>
      <th>139</th>
      <td>l'escalade</td>
      <td>Dugomier</td>
      <td>2</td>
    </tr>
    <tr>
      <th>146</th>
      <td>la chaleur</td>
      <td>Victor Jestin</td>
      <td>10</td>
    </tr>
    <tr>
      <th>150</th>
      <td>la colère</td>
      <td>S. A. Cosby</td>
      <td>8</td>
    </tr>
    <tr>
      <th>159</th>
      <td>la décision</td>
      <td>Tuil,  Karine</td>
      <td>9</td>
    </tr>
    <tr>
      <th>160</th>
      <td>la dépendance</td>
      <td>Rachel Cusk</td>
      <td>3</td>
    </tr>
    <tr>
      <th>161</th>
      <td>la faille</td>
      <td>Franck Thilliez</td>
      <td>2</td>
    </tr>
    <tr>
      <th>171</th>
      <td>la grâce</td>
      <td>Thibault de Montaigu</td>
      <td>13</td>
    </tr>
    <tr>
      <th>173</th>
      <td>la maison</td>
      <td>Emma Becker</td>
      <td>77</td>
    </tr>
    <tr>
      <th>193</th>
      <td>la tentation</td>
      <td>Luc Lang</td>
      <td>1</td>
    </tr>
    <tr>
      <th>195</th>
      <td>la vie nouvelle</td>
      <td>Tom Crewe</td>
      <td>1</td>
    </tr>
    <tr>
      <th>198</th>
      <td>la vraie vie</td>
      <td>Adeline Dieudonné</td>
      <td>10</td>
    </tr>
    <tr>
      <th>202</th>
      <td>le bon sens</td>
      <td>Michel Bernard</td>
      <td>7</td>
    </tr>
    <tr>
      <th>207</th>
      <td>le consentement</td>
      <td>Vanessa Springora</td>
      <td>1</td>
    </tr>
    <tr>
      <th>209</th>
      <td>le culte</td>
      <td>Camilla Läckberg, Henrik Fexeus</td>
      <td>2</td>
    </tr>
    <tr>
      <th>215</th>
      <td>le droit du sol</td>
      <td>Étienne Davodeau</td>
      <td>1</td>
    </tr>
    <tr>
      <th>219</th>
      <td>le gosse</td>
      <td>Olmi,  Véronique</td>
      <td>2</td>
    </tr>
    <tr>
      <th>227</th>
      <td>le lâche</td>
      <td>Jarred McGinnis</td>
      <td>7</td>
    </tr>
    <tr>
      <th>236</th>
      <td>le retournement</td>
      <td>Manuel Carcassonne</td>
      <td>1</td>
    </tr>
    <tr>
      <th>262</th>
      <td>les cinq</td>
      <td>Mathieu Rochelle</td>
      <td>8</td>
    </tr>
    <tr>
      <th>267</th>
      <td>les dragons</td>
      <td>Jérôme Colin</td>
      <td>3</td>
    </tr>
    <tr>
      <th>282</th>
      <td>les impatientes</td>
      <td>Djaïli Amadou Amal</td>
      <td>1</td>
    </tr>
    <tr>
      <th>288</th>
      <td>les passagers</td>
      <td>Gilles Verdet</td>
      <td>1</td>
    </tr>
    <tr>
      <th>293</th>
      <td>les possibles</td>
      <td>Grimaldi,  Virginie</td>
      <td>1</td>
    </tr>
    <tr>
      <th>348</th>
      <td>nature humaine</td>
      <td>Serge Joncour</td>
      <td>4</td>
    </tr>
    <tr>
      <th>351</th>
      <td>noa</td>
      <td>Lévy,  Marc</td>
      <td>4</td>
    </tr>
    <tr>
      <th>366</th>
      <td>otages</td>
      <td>Nina Bouraoui</td>
      <td>4</td>
    </tr>
    <tr>
      <th>370</th>
      <td>panorama</td>
      <td>Lilia Hassaine</td>
      <td>1</td>
    </tr>
    <tr>
      <th>371</th>
      <td>papa</td>
      <td>Hélène Delforge et Quentin Gréban</td>
      <td>61</td>
    </tr>
    <tr>
      <th>372</th>
      <td>par les routes</td>
      <td>Sylvain Prudhomme</td>
      <td>1</td>
    </tr>
    <tr>
      <th>377</th>
      <td>performance</td>
      <td>Simon Liberati</td>
      <td>18</td>
    </tr>
    <tr>
      <th>399</th>
      <td>révolution</td>
      <td>Florent Grouazel</td>
      <td>30</td>
    </tr>
    <tr>
      <th>400</th>
      <td>s'adapter</td>
      <td>Clara Dupont-Monod</td>
      <td>9</td>
    </tr>
    <tr>
      <th>401</th>
      <td>sa préférée</td>
      <td>Sarah Jollien-Fardel</td>
      <td>1</td>
    </tr>
    <tr>
      <th>402</th>
      <td>sabre</td>
      <td>Emmanuel Ruben</td>
      <td>4</td>
    </tr>
    <tr>
      <th>423</th>
      <td>tenir sa langue</td>
      <td>Polina Panassenko</td>
      <td>1</td>
    </tr>
    <tr>
      <th>443</th>
      <td>tueurs</td>
      <td>Jean-Michel Espitallier</td>
      <td>1</td>
    </tr>
    <tr>
      <th>452</th>
      <td>un miracle</td>
      <td>Mas,  Victoria</td>
      <td>2</td>
    </tr>
    <tr>
      <th>472</th>
      <td>vivonne</td>
      <td>Jérôme Leroy</td>
      <td>1</td>
    </tr>
    <tr>
      <th>482</th>
      <td>à la ligne</td>
      <td>Joseph Ponthus</td>
      <td>1</td>
    </tr>
    <tr>
      <th>485</th>
      <td>à tout jamais</td>
      <td>Colleen Hoover</td>
      <td>1</td>
    </tr>
  </tbody>
</table>
</div>

In [ ]:
df = livre_dans_sous_titres('Base_finale','ss_youtube_nouveauté_livre_2023')
df.shape # connaitre le nombre de lignes de la base de données obtenue

**45 livres apparaissent dans les sous-titres des vidéos youtube.** Le nombre d'occurrences varie entre 1 et 127

# 2 Recherche des auteurs de livres dans les descriptions de vidéos youtube

In [26]:
#On importe la bibliothèque pandas
import pandas as pd
import matplotlib.pyplot as plt

def auteur_dans_sous_titres(livre,youtube):
    #On charge les tables dont on a besoin",
    books_data = pd.read_csv(f"{livre}.csv", encoding="utf-8")
    youtube_data = pd.read_csv(f"{youtube}.csv", encoding="utf-8")
    print("Auteurs de la base de données "f'{livre}.csv'" dans les sous-titres de vidéos youtube de la base de données "f'{youtube}.csv')
    # On convertie la colonne 'Subtitles'et 'Auteur' en type string
    youtube_data['Subtitles'] = youtube_data['Subtitles'].astype(str)
    books_data['Auteur'] = books_data['Auteur'].astype(str)
    # On initialise une nouvelle liste pour stocker les résultats
    result = []
    # On itére sur chaque liste de la colonne 'Auteur' de books_data
    for index1, row1 in books_data.iterrows():
        count_list = []  # Liste pour stocker les résultats de cette ligne
        # On itére sur chaque liste de la colonne 'Subtitles' de youtube_data
        for index2, row2 in youtube_data.iterrows():
            # On vérifie si la liste de books_data (row1['Auteur']) est une sous-liste de la liste de youtube_data (row2['Subtitles'])
            if row1['Auteur'] in row2['Subtitles']:
                count_list.append(1)  # Si elle est incluse, compter une occurrence
            else:
                count_list.append(0)  # Sinon, aucune occurrence
        # On ajoute la liste des Auteurs de books_data et la somme des occurrences
        result.append([row1['Auteur'], sum(count_list)])
    # On crée la nouvelle DataFrame contenant les résultats
    colonnes=['Auteurs','Occurrences dans les sous-titres de vidéos']
    result_df = pd.DataFrame(result, columns = colonnes )
    # On filtre les lignes où 'Occurrences dans les sous-titres des vidéos' est différente de 0 (On ne s'intéresse qu'aux livres cités au moins une fois)
    filtered_df = result_df[result_df['Occurrences dans les sous-titres de vidéos'] != 0]
    # On affiche le DataFrame filtré
    return filtered_df

def auteur_dans_sous_titres2(livre,youtube):
    #On charge les tables dont on a besoin",
    books_data = pd.read_csv(f"{livre}.csv", encoding="utf-8")
    youtube_data = pd.read_csv(f"{youtube}.csv", encoding="utf-8")
    print("Auteurs de la base de données "f'{livre}.csv'" dans les sous-titres de vidéos youtube de la base de données "f'{youtube}.csv')
    # On convertie la colonne 'Subtitles'et 'auteur' en type string
    youtube_data['Subtitles'] = youtube_data['Subtitles'].astype(str)
    books_data['auteur'] = books_data['auteur'].astype(str)
    # On initialise une nouvelle liste pour stocker les résultats
    result = []
    # On itére sur chaque liste de la colonne 'auteur' de books_data
    for index1, row1 in books_data.iterrows():
        count_list = []  # Liste pour stocker les résultats de cette ligne
        # On itére sur chaque liste de la colonne 'Subtitles' de youtube_data
        for index2, row2 in youtube_data.iterrows():
            # On vérifie si la liste de books_data (row1['auteur']) est une sous-liste de la liste de youtube_data (row2['Subtitles'])
            if row1['auteur'] in row2['Subtitles']:
                count_list.append(1)  # Si elle est incluse, compter une occurrence
            else:
                count_list.append(0)  # Sinon, aucune occurrence
        # On ajoute la liste des auteurs de books_data et la somme des occurrences
        result.append([row1['auteur'], sum(count_list)])
    # On crée la nouvelle DataFrame contenant les résultats
    colonnes=['Auteurs','Occurrences dans les sous-titres de vidéos']
    result_df = pd.DataFrame(result, columns = colonnes )
    # On filtre les lignes où 'Occurrences dans les sous-titres des vidéos' est différente de 0 (On ne s'intéresse qu'aux livres cités au moins une fois)
    filtered_df = result_df[result_df['Occurrences dans les sous-titres de vidéos'] != 0]
    # On affiche le DataFrame filtré
    return filtered_df 

## 2.1 Recherche des auteurs de livre dans la base de données youtube_livre_2019_2023.csv

### 2.1.1 Recherche des auteurs de livre de la base de données 23_livres_les_plus_lus_babelio.csv


In [ ]:
auteur_dans_sous_titres('23_livres_les_plus_lus_babelio','ss_youtube_livre_2019_2023')

**Les auteurs Virginie Grimaldi et Stephen King apparaissent chacun dans les sous-titres d'une vidéo youtube**. Ces deux auteurs qui ont écrit chacun un livre parmi les plus lus en 2023 sur Babelio sont chacun dans les sous-titres d'une vidéo youtube.

### 2.1.2 Recherche des auteurs de livre de la base de données best_sellers_fnac_2023_cleaned.csv


In [32]:
auteur_dans_sous_titres('best_sellers_fnac_2023_cleaned','ss_youtube_livre_2019_2023')

Auteurs de la base de données best_sellers_fnac_2023_cleaned.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
22  Éric-Emmanuel Schmitt                                           1
24           Stephen King                                           1
28      Virginie Grimaldi                                           1
52                    Léo                                           2
61         Britney Spears                                           1

**L'auteur Léo qui a écrit en 2023 un best_sellers de la Fnac apparaît dans la description youtube d'une vidéo.**

### 2.1.4 Recherche des auteurs de livre de la base de données Les livres les plus empruntés à Paris.csv


In [34]:
auteur_dans_sous_titres2('livres_les_plus_empruntes_a_paris','ss_youtube_livre_2019_2023')

Auteurs de la base de données livres_les_plus_empruntes_a_paris.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
75     nan                                         158

"nan" n'est pas un auteur mais simplement la dénomination utilisée dans la base de données lorsqu'on connaît pas l'auteur. Donc il n'y a aucun auteur de cette base de données mentionné.

### 2.1.5 Recherche des auteurs de livre de la base de données livraddict_prix_2024.csv


In [35]:
auteur_dans_sous_titres('livraddict_prix_2024','ss_youtube_livre_2019_2023')

Auteurs de la base de données livraddict_prix_2024.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
13  Virginie Grimaldi                                           1

Il n'y a aucun auteur de cette base de données mentionné.

### 2.1.6 Recherche des auteurs de livre de la base de données prix_litteraires.csv

In [37]:
auteur_dans_sous_titres('prix_litteraires','ss_youtube_livre_2019_2023')

Auteurs de la base de données prix_litteraires.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
16    Copi                                           1

**L'auteur Joseph Ponthus qui a obtenu un prix littéraires pour un livre apparaît dans la description youtube d'une vidéo.**

### 2.1.7 Recherche des auteurs de livre de la base de données Base_finale.csv

In [38]:
auteur_dans_sous_titres('Base_finale','ss_youtube_livre_2019_2023')

Auteurs de la base de données Base_finale.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_livre_2019_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
22                     Léo                                           2
61            Stephen King                                           1
163         Britney Spears                                           1
201                   Copi                                           1
216  Éric-Emmanuel Schmitt                                           1
379                    nan                                         158
458      Virginie Grimaldi                                           1

**Les auteurs Léo et Joseph Ponthus apparaissent chacun dans une description de vidéo youtube.** "nan" n'est pas un auteur mais simplement la dénomination utilisée dans la base de données lorsqu'on connaît pas l'auteur. Donc il n'y a aucun auteur de cette base de données mentionné.

## 2.2 Recherche des auteurs de livre dans la base de données youtube_nouveauté_livre_2023.csv

### 2.2.1 Recherche des auteurs de livre de la base de données 23 livres les plus lus Babelio.csv


In [39]:
auteur_dans_sous_titres('23_livres_les_plus_lus_babelio','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données 23_livres_les_plus_lus_babelio.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
15  Michel Bussi                                           1
18  Stephen King                                           6

**Les auteurs Michel Bussi et Virginie Grimaldi qui ont chacun écrit un des 23 livres les plus lus sur Babelio apparaissent chacun dans une description de vidéo youtube.**

### 2.2.2 Recherche des auteurs de livre de la base de données best_sellers_fnac_2023_cleaned.csv


In [40]:
auteur_dans_sous_titres('best_sellers_fnac_2023_cleaned','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données best_sellers_fnac_2023_cleaned.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
15  Michel Bussi                                           1
24  Stephen King                                           6
34      Squeezie                                           1
44  Hannah Grace                                           1
52           Léo                                          17

**Les auteurs Michel Bussi, Virginie Grimaldi et Agnès Martin-Lugand apparaissent chacun dans un description de vidéo youtube et Léo apparaît à deux reprise.** Ce sont tous des auteurs qui ont écrit un best seller de la Fnac en 2023.

### 2.2.4 Recherche des auteurs de livre de la base de données Les livres les plus empruntés à Paris.csv


In [41]:
auteur_dans_sous_titres2('livres_les_plus_empruntes_a_paris','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données livres_les_plus_empruntes_a_paris.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
75     nan                                         264

"nan" n'est pas un auteur mais simplement la dénomination utilisée dans la base de données lorsqu'on connaît pas l'auteur. Donc il n'y a aucun auteur de cette base de données mentionné.

### 2.2.5 Recherche des auteurs de livre de la base de données livraddict_prix_2024.csv


In [42]:
auteur_dans_sous_titres('livraddict_prix_2024','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données livraddict_prix_2024.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Empty DataFrame
Columns: [Auteurs, Occurrences dans les sous-titres de vidéos]
Index: []

**L'autrice Virginie Grimaldi qui a écrit un des livres les mieux notés par les utilisateurs de livraddict parmi les livres sortis en 2023, apparaît dans une description de vidéo youtube.**

### 2.2.6 Recherche des auteurs de livre de la base de données prix_litteraires.csv

In [43]:
auteur_dans_sous_titres('prix_litteraires','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données prix_litteraires.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
16    Copi                                           2

Il n'y a aucun auteur de cette base de données mentionné.

### 2.2.7 Recherche des auteurs de livre de la base de données Base_finale.csv

In [44]:
auteur_dans_sous_titres('Base_finale','ss_youtube_nouveauté_livre_2023')

Auteurs de la base de données Base_finale.csv dans les sous-titres de vidéos youtube de la base de données ss_youtube_nouveauté_livre_2023.csv


Auteurs  Occurrences dans les sous-titres de vidéos
22            Léo                                          17
30       Squeezie                                           1
61   Stephen King                                           6
118  Hannah Grace                                           1
201          Copi                                           2
379           nan                                         264
441  Michel Bussi                                           1

**Les auteurs Léo, Agnès Martin-Lugand, Michel Bussi et Virginie Grimaldi apparaissent dans des descriptions vidéos.** "nan" n'est pas un auteur mais simplement la dénomination utilisée dans la base de données lorsqu'on connaît pas l'auteur. Donc il n'y a aucun auteur de cette base de données mentionné.

# Conclusion

En résumé, en tout **8 livres** (Les livres *Babel* de R. F. Kuang,  *À la ligne* de Joseph Ponthus, *Troies vies par semaine* de Michel Bussi, *Veiller sur elle* de Jean-Baptiste Andrea, *Le culte* de Camilla Läckberg et Henrik Fexeus, *Otages* de Nina Bouraoui, *Performance* de Simon Liberati et *Révolution* de Florent Grouazel) et **5 auteurs** (Joseph Ponthus, Léo, Michel Bussi, Virginie Grimaldi et Agnès Martin-Lugand) apparaissent dans des descriptions de vidéos youtube des deux bases de données utilisées.
Cela laisserait donc penser que seulement 8 livres et 5 auteurs apparaissent dans les vidéos youtubes de nos bases de données.
Cependant plusieurs points sont à prendre en compte:
-**des vidéos youtubes peuvent être sur un livre ou un auteur sans le citer dans la description, il peut aussi ne même pas y avoir de description.**
-**il peut y avoir des fautes d'orthographes et des erreurs de ponctuation dans les titres des livres dans les description youtube qui ne sont alors pas comptés dans ce programme.**
On peut pour autant remarquer que **certains titres de livre ou noms d'auteur sont assez courants pour désigner autre chose, ce qui peut fausser les résultats.**

**Pour essayer d'obtenir de meilleurs résultats, on va par la suite rechercher les titres et auteurs des livres dans les sous-titres de vidéos des bases de données en possédant dans le fichier stat_descriptives_youtube.ipynb**